pkg/action/reset.go

```diff
@@ -146,30 +157,34 @@ func (r ResetAction) Run() (err error) {
 		oem := r.spec.Partitions.OEM
 		if oem != nil {
 			// Try to umount
-			err = e.UnmountPartitions(r.spec.Partitions.PartitionsByMountPoint(true, oem))
+			err = e.UnmountPartition(oem)
 			if err != nil {
 				return err
 			}
 			err = e.FormatPartition(oem)
 			if err != nil {
 				return err
 			}
+			// Mount it back, as oem is mounted during recovery, keep everything as is
+			err = e.MountPartition(oem)
+			if err != nil {
+				return err
+			}
 		}
 	}
-	// Mount configured partitions
-	err = e.MountPartitions(r.spec.Partitions.PartitionsByMountPoint(false, r.spec.Partitions.Recovery))
+
+	// Before reset hook happens once partitions are aready and before deploying the OS image
+	err = r.resetHook(cnst.BeforeResetHook, false)
```

Comment:
any reason why we are changing the logic here? What if the before-reset hook need access to the state partition? It used to have access until now (since we mounted first)

Reply:
I dont think we should mount everything like crazy here. If the reset hook needs persistent they should mount them in there, like we do in other places like oem and such. Recovery should only have recovery and oem mounted at teh start and then leave them mounted at the end.

Reply:
If you check the hooks, you can see for example that we assume that nothing is mounted and mount things explicitely and then unmount them. I think thats the correct approach, especially on recovery where we dont want to leave things around mounted.